embeddings.mdx

```diff
@@ -0,0 +1,147 @@
+    )
+```
+
+### Instructor
+
+Chalk supports [INSTRUCTOR](https://github.com/xlang-ai/instructor-embedding) embedding models.
+The default model for this provider is `hkunlp/instructor-base`. The full list of models is available
+[here](https://github.com/xlang-ai/instructor-embedding#model-list).
+
+```python
+from chalk.features import embedding, features, Vector
+
+@features
+class Document:
+    content: str
+    embedding: Vector = embedding(
+        feature="content",
+        provider="instructor",
+        model="hkunlp/instructor-base",  # optional, defaults to hkunlp/instructor-base
+    )
+```
+
+### OpenAI
+
+Chalk can proxy calls to the [OpenAI Embeddings API](https://platform.openai.com/docs/guides/embeddings).
+The default model for this provider is `text-embedding-ada-002`.
+To use OpenAI, first sign up for an
+[OpenAI Account](https://openai.com/api/), and then create an [OpenAI Integration](/docs/openai) in Chalk.
+All OpenAI requests will be attributed to your API key. To minimize usage, we highly recommend specifying an
+appropriate TTL in Chalk, which will ensure that embeddings are cached.
+
+```python
+from chalk.features import embedding, features, Vector
+
+@features
+class Document:
+    content: str
+    embedding: Vector = embedding(
+        feature="content",
+        provider="openai",
+        model="text-embedding-ada-002",  # optional, defaults to text-embedding-ada-002
+        ttl="infinity",
+    )
+```
+
+### Cohere
+
+Chalk can proxy calls to [Cohere Embed](https://cohere.com/embed). The default model for this provider is `embed-english-v2.0`.
+To use this integration, first sign up for an
+[Cohere Account](https://dashboard.cohere.com/welcome/register), and then create an [Cohere Integration](/docs/cohere) in Chalk.
+All Cohere requests will be attributed to your API key. To minimize usage, we highly recommend specifying an
+appropriate TTL in Chalk, which will ensure that embeddings are cached.
+
+```python
+from chalk.features import embedding, features, Vector
+
+@features
+class Document:
+    content: str
+    embedding: Vector = embedding(
+        feature="content",
+        provider="cohere",
+        model="embed-english-v2.0",  # optional, defaults to embed-english-v2.0
+        ttl="infinity",
+    )
+```
+
+## Custom embedding functions
+
+If you would like to run your own embedding model, you can define a custom resolver to compute the embedding
+from existing features in the feature set. For performance, we recommend to store the model weights in an
+object store (such as AWS S3 or GCS) rather than including them your source code and to load the model
+using a [boot hook](/docs/generic#initializing).
+
+```python
+from chalk.features import before_all, DataFrame, embedding, features, realtime, Vector
+
+my_model = None
+
+@before_all
+def load_my_model():
+    global my_model
+    my_model = MyModel().initialize("s3://my-bucket/my-checkpoint.pt")
```

Comment:
I agree this is fine, but it might read nicer to folks who don't know about global as 

```suggestion
my_model = MyModel()

@before_all
def load_my_model():
    my_model.initialize("s3://my-bucket/my-checkpoint.pt")
```

Reply:
In practice you wouldn't want to initialize your model on import time... but sure, definitely simplies it for this example.